apis/python/src/tiledbsoma/io/registration/id_mappings.py

```diff
@@ -0,0 +1,81 @@
+from typing import Dict, List
+
+import anndata as ad
+import attrs
+import pandas as pd
+from typing_extensions import Self
+
+import tiledbsoma
+import tiledbsoma.logging
+
+
+@attrs.define(kw_only=True)
+class AxisIDMapping:
+    """
+    For a single to-be-appended AnnData/H5AD input in SOMA multi-file append-mode ingestion, this
+    class tracks the mapping of input-data ``obs`` or ``var`` 0-up offsets to SOMA join ID values
+    for the destination SOMA experiment.
+
+    See module-level comments for more information.
+    """
+
+    data: List[int]
```

Comment:
Do we expect to be mutating this object during the course of an ingestion (i.e., changing the contents of this list, as opposed to creating a new `AxisIDMapping` with different contents)? If not, it might be worth making this a `Tuple[int, ...]` to avoid accidental spooky action at a distance (e.g. somebody passes `whatever.data` into a method that mutates it).